static_frame/core/frame.py

```diff
@@ -1514,21 +1516,36 @@ def from_sql(cls,
             if index_depth == 1:
                 index_constructor = Index
 
-                def row_gen_final() -> tp.Iterator[tp.Sequence[tp.Any]]:
-                    for row in row_gen:
+                def row_gen() -> tp.Iterator[tp.Sequence[tp.Any]]:
+                    for row in sql_iterable:
                         index.append(row[0])
                         yield row[1:]
 
             else: # > 1
                 index_constructor = IndexHierarchy.from_labels
 
-                def row_gen_final() -> tp.Iterator[tp.Sequence[tp.Any]]:
-                    for row in row_gen:
+                def row_gen() -> tp.Iterator[tp.Sequence[tp.Any]]:
+                    for row in sql_iterable:
                         index.append(row[:index_depth])
                         yield row[index_depth:]
         else:
             index = None
-            row_gen_final = lambda: row_gen
+            row_gen = lambda: sql_iterable
+
+        if columns_select:
+            columns_select = set(columns_select)
+
+            sel = columns.isin(columns_select)
+            # desired_cols = columns.loc_to_iloc(sel) # Currently doesn't work for IH...
+            desired_cols = np.where(sel)[0]
```

Comment:
I see that a Boolean array given to `loc_to_iloc` does not give the expected result; let me see if I can fix that, as having to use `np.where` here seems undesirable. I will try to push a fix for this later today.